users/views.py

```diff
@@ -1,3 +1,93 @@
-from django.shortcuts import render
+import json, bcrypt, jwt, re
 
-# Create your views here.
+from django.http    import JsonResponse
+from django.views   import View
+
+from .models        import User
+import my_settings
+
+
+class SignUpView(View):
+    
+    def post(self, request):
+        
+        data = json.loads(request.body)
+        MINIMUM_PASSWORD_LENGTH = 8
+        MINIMUM_ACCOUNT_LENGTH  = 5
+
+        try:
+            account       = data['account']
+            division      = data['division']
+            password      = data['password']
+            name          = data['name']
+            phone_number  = data['phone_number']
+            email         = data['email']
+            date_of_birth = data['date_of_birth']
+            gender        = data['gender']
+
+            email_check = re.compile('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+            email_check = email_check.match(email)
+
+            if len(account) < MINIMUM_ACCOUNT_LENGTH:
+                return JsonResponse({'MESSAGE' : 'INVALID_ACCOUNT'}, status = 400)
+
+            if User.objects.filter(account=account).exists():
+                return JsonResponse({'MESSAGE' : 'DUPLICATE_ACCOUNT'}, status = 400)
+
+            if User.objects.filter(phone_number=phone_number).exists():
+                return JsonResponse({'MESSAGE' : 'DUPLICATE_PHONE_NUMBER'}, status = 400)
+
+            if email_check == None:
+                return JsonResponse({'MESSAGE' : 'INVALID_EMAIL'}, status = 400)
+
+            if len(password) < MINIMUM_PASSWORD_LENGTH:
+                return JsonResponse({'MESSAGE' : 'INVALID_PASSWORD'}, status = 400)
+
+            hashed_password = bcrypt.hashpw(
+                    password.encode('utf-8'), 
+                    bcrypt.gensalt()
+                    )
+
+            User.objects.create(
+                division      = data['division'],
```

Comment:
이미 위에서 변수처리 한 변수들을 사용하셔도 됩니다!
```suggestion
                division      = division,
```

Reply:
네, 선언된 변수를 활용하는 것으로 수정했습니다!